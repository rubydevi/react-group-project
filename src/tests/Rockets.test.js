import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; // Import mock store
import RocketsComponent from '../components/Rockets';

// Create a mock store with the state you want to use in the test
const mockStore = configureStore([]);

test('RocketsComponent renders correctly', () => {
  const store = mockStore({
    rockets: {
      rockets: [
        {
          id: '5e9d0d95eda69955f709d1eb',
          name: 'Falcon 1',
          description: 'Description 1',
          flickr_images: ['https://imgur.com/DaCfMsj.jpg'],
        },
        {
          id: '5e9d0d95eda69973a809d1ec',
          name: 'Falcon 9',
          description: 'Description 2',
          flickr_images: ['https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg'],
        },
      ],
      loading: false,
      error: null,
    },
  });

  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <RocketsComponent />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
