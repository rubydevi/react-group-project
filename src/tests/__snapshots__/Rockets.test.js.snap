// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketsComponent renders correctly 1`] = `
<div
  className="container"
>
  <ul
    className="rockets"
  >
    <li
      className="rocket"
    >
      <img
        alt="Rocket Falcon 1"
        className="rocket-image"
        src="https://imgur.com/DaCfMsj.jpg"
      />
      <div
        className="rocket-details"
      >
        <h3
          className="rocket-title"
        >
          Falcon 1
        </h3>
        <div>
          <p
            className="rocket-description"
          >
            Description 1
          </p>
        </div>
        <button
          className="reserve-button"
          onClick={[Function]}
          type="submit"
        >
          Reserve Rocket
        </button>
      </div>
    </li>
    <li
      className="rocket"
    >
      <img
        alt="Rocket Falcon 9"
        className="rocket-image"
        src="https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg"
      />
      <div
        className="rocket-details"
      >
        <h3
          className="rocket-title"
        >
          Falcon 9
        </h3>
        <div>
          <p
            className="rocket-description"
          >
            Description 2
          </p>
        </div>
        <button
          className="reserve-button"
          onClick={[Function]}
          type="submit"
        >
          Reserve Rocket
        </button>
      </div>
    </li>
  </ul>
</div>
`;
